### ISSUE TITLE
Tenant-specific deployment targets view

### ISSUE THREAD

**Issue by michaelnoonan** [OctopusDeploy/Issues#2737]
![image](https://cloud.githubusercontent.com/assets/1627582/18369845/8f9484d8-766b-11e6-9b84-626ba811a587.png)

I’ve been thinking a bit about Deployment Targets + Tenants also, and I’m starting to lean towards having sense of ownership:

1. Create a deployment target from the tenant screen = target is ​**owned**​ by the Tenant
   
   a. You cannot scope other tenants to that deployment target - much safer for the “dedicated” scenario
   b. Deleting the tenant would have a different impact on the target

2. Create a deployment target in the environment screen = target is not ​**owned**​

   a. You can scope the deployment target to one or more tenants
   b. Deleting a tenant simply removes them from any target scoping, even if that would make the target empty

This would cater for one of the original requests: https://octopus.com/blog/rfc-multitenancy#comment-2402945580

And some of the comments from here: http://help.octopusdeploy.com/discussions/beta-testing-feedback/77#comment_39924247

**Comment by rheap** [comment]
I for one would love to see this implemented! It would help closer tie targets and tenants together and make managing tenants easier for us as all our tenants have dedicated machines.

**Comment by michaelnoonan** [comment]
In Octopus 4.0 you can search/filter by tenants in the Infrastructure/Environments page:

![image](https://user-images.githubusercontent.com/1627582/31409070-ee4de152-ae4d-11e7-8c87-4e1f3f184392.png)

I wonder if this is enough, or whether we should try and rehost the deployment targets list inside the Tenant as well? @thebigmoosey what do you think?

**Comment by BlairPaine** [comment]
@michaelnoonan Exceptional! I'm so very excited to see this will be in Octopus 4.0 as it has been one of the big downsides for us in using Octopus Deploy in a large environment.  With many machines and many tenants the environments page was becoming less friendly.  I would love to be a beta tester and provide feedback.

It would be fantastic to also be able to click on a Tenant on the tenants page and see the view of deployment targets assigned to them for each environment or even just link to a filter of the environments page.

**Comment by michaelnoonan** [comment]
Hi @BlairPaine - great news! I think you'll really like the new "Infrastructure" and "Environments" areas. Jump on board over here: https://twitter.com/OctopusDeploy/status/918039069040324608

**Comment by thebigmoosey** [comment]
@michaelnoonan Yes good idea! I think a deployment targets section in the Tenants area will be something we can investigate ... will put it on the ToDo list to look into now.

**Comment by BlairPaine** [comment]
Hi @thebigmoosey - is there an open issue for having deployment targets added into the Tenants area I can track/follow please?

**Comment by thebigmoosey** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Deployment targets in tenants sidebar](https://trello.com/c/aiNUFKyK/471-deployment-targets-in-tenants-sidebar)

**Comment by thebigmoosey** [comment]
Hi @BlairPaine - we'll keep this issue (#2737) open until we've implemented this feature into the Tenants area.
